### PR TITLE
feat(API): add ready state checks and all automated tests #38 #75

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #38, #75

## Summary:

Adds the ready check behavior for peer dependencies and all automated testing to pass threshold requirements.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/auro-select.js [hunk 1]
```diff
@@ -128,32 +165,95 @@ class AuroSelect extends LitElement {
...
+  }
 
-    // custom event listener from auro-menu fires with both 'click' and keypress events
-    this.addEventListener('selectedOption', (evt) => {
-      const dropdown = this.shadowRoot.querySelector('auro-dropdown');
-      const triggerContentEl = dropdown.querySelector('#triggerFocus');
+  /**
+   * @private
+   * @returns {void} Marks the component as ready and sends event.
+   */
```
Comment (blackfalcon) [inline src/auro-select.js:173]: Please follow the preferred JSDoc pattern. 

```
  /**
   * Marks the component as ready and sends event.
   * @private
   * @returns {void} 
   */
```

File: src/auro-select.js [hunk 2]
```diff
@@ -128,32 +165,95 @@ class AuroSelect extends LitElement {
...
+  readyActions() {
+    // Set the initial value in auro-menu if defined
+    if (this.hasAttribute('value') && this.getAttribute('value').length > 0) {
+      this.menu.value = this.value;
+    }
+  }
+
+  /**
+   * @private
+   */
```
Comment (blackfalcon) [inline src/auro-select.js:228]: Please add the appropriate contextual information for this comment header.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
I would have preferred that the different updates were kept in individual commits, e.g. tests versus features. 

Other than that, please update the JSDoc comments are you are good to go.

**Review by MattSidor — APPROVED** [review]

**Pull request merged**